File: implementations/node/core/years/2022/day-7/part-2.mjs
import { dirSizes } from "./common.mjs";

const sizeOfDisk = 70_000_000
const needed = 30_000_000

function smallestDirToMakeEnoughFreeSpace(dirs) {
    const free = sizeOfDisk - dirs['/']
    return Math.min(...Object.values(dirs).filter(value => free + value > needed))
}
export function execute(input) {
    return smallestDirToMakeEnoughFreeSpace(dirSizes(input))
}

File: implementations/rust/src/years/y2022/day_7/part_2.rs
use super::common::collect_dirs;

pub fn execute(input: String) -> String {
    let mut dirs = collect_dirs(input);
    let empty_space = 70_000_000 - dirs.get(&String::from("/")).unwrap();
    dirs.values().filter(|size| empty_space + **size > 30_000_000).min().unwrap().to_string()
}